src/planner/parabolic/ParabolicUtil.cpp

```diff
@@ -0,0 +1,230 @@
+
+  // Construct a list of all ramp transition points.
+  double t = 0.;
+  std::set<double> transitionTimes;
+  transitionTimes.insert(t);
+
+  for (const auto& rampNd : _inputPath.ramps)
+  {
+    for (const auto& ramp1d : rampNd.ramps)
+    {
+      transitionTimes.insert(t + ramp1d.tswitch1);
+      transitionTimes.insert(t + ramp1d.tswitch2);
+    }
+
+    t += rampNd.endTime;
+    transitionTimes.insert(t);
+  }
+
+  // Convert the output to a spline with a knot at each transition time.
+  assert(!transitionTimes.empty());
+  const auto startIt = std::begin(transitionTimes);
+  double timePrev = *startIt;
+  transitionTimes.erase(startIt);
+
+  Eigen::VectorXd positionPrev, velocityPrev;
+  evaluateAtTime(_inputPath, timePrev, positionPrev, velocityPrev);
+
+  auto _outputTrajectory = make_unique<aikido::trajectory::Spline>(
+      _stateSpace, timePrev + _startTime);
+  auto segmentStartState = _stateSpace->createState();
+
+  for (const auto timeCurr : transitionTimes)
+  {
+    Eigen::VectorXd positionCurr, velocityCurr;
+    evaluateAtTime(_inputPath, timeCurr, positionCurr, velocityCurr);
+
+    CubicSplineProblem problem(Vector2d(0, timeCurr - timePrev), 4, dimension);
+    problem.addConstantConstraint(0, 0, Eigen::VectorXd::Zero(dimension));
+    problem.addConstantConstraint(0, 1, velocityPrev);
+    problem.addConstantConstraint(1, 0, positionCurr - positionPrev);
+    problem.addConstantConstraint(1, 1, velocityCurr);
+    const auto spline = problem.fit();
+
+    _stateSpace->expMap(positionPrev, segmentStartState);
+
+    // Add the ramp to the output trajectory.
+    assert(spline.getCoefficients().size() == 1);
+    const auto& coefficients = spline.getCoefficients().front();
+    _outputTrajectory->addSegment(
+        coefficients, timeCurr - timePrev, segmentStartState);
+
+    timePrev = timeCurr;
+    positionPrev = positionCurr;
+    velocityPrev = velocityCurr;
+  }
+
+  return _outputTrajectory;
+}
+
+std::unique_ptr<ParabolicRamp::DynamicPath> convertToDynamicPath(
+    const aikido::trajectory::Spline& _inputTrajectory,
+    const Eigen::VectorXd& _maxVelocity,
+    const Eigen::VectorXd& _maxAcceleration,
+    bool _useVelocity)
+{
+  const auto stateSpace = _inputTrajectory.getStateSpace();
+  const auto numWaypoints = _inputTrajectory.getNumWaypoints();
+
+  std::vector<ParabolicRamp::Vector> milestones;
+  std::vector<ParabolicRamp::Vector> velocities;
+  milestones.reserve(numWaypoints);
+  velocities.reserve(numWaypoints);
+
+  Eigen::VectorXd tangentVector, currVec;
+
+  for (size_t iwaypoint = 0; iwaypoint < numWaypoints; ++iwaypoint)
+  {
+    auto currentState = stateSpace->createState();
+    _inputTrajectory.getWaypoint(iwaypoint, currentState);
+
+    stateSpace->logMap(currentState, currVec);
+    milestones.emplace_back(toVector(currVec));
+
+    _inputTrajectory.getWaypointDerivative(iwaypoint, 1, tangentVector);
+    velocities.emplace_back(toVector(tangentVector));
```

Comment:
Resolved by splitting the `Spline` and `Interpolated` cases into separate overloads. 👍